src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/LaunchProfilesDebugLaunchProvider.cs

```diff
@@ -74,7 +75,7 @@ private async Task<ILaunchProfile> GetActiveProfileAsync()
         {
             // Get the active debug profile (timeout of 5s, though in reality is should never take this long as even in error conditions
             // a snapshot is produced).
-            ILaunchSettings? currentProfiles = await _launchSettingsProvider.WaitForFirstSnapshot(5000);
+            ILaunchSettings? currentProfiles = await _launchSettingsProvider.WaitForFirstSnapshot(Timeout.Infinite);
```

Comment:
A suggestion for future: It seems that most invocations of this method pass an infinite timeout. Perhaps an overload or extension method that takes no arguments, implying an infinite timeout, would discourage future code from putting some arbitrary value in here.

Reply:
Long-term I think we need a replacement that takes a `CancellationToken` and then we deprecate this UI.

Reply:
It's part of our public API surface area, so we can't remove it easily.